src/ForceTorqueMonitor.c

```diff
@@ -0,0 +1,164 @@
+// ForceTorqueMonitor.c
+
+// SPDX-FileCopyrightText: 2022-2023, Yaskawa America, Inc.
+// SPDX-FileCopyrightText: 2022-2023, Delft University of Technology
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotoROS.h"
+
+ForceTorqueMonitor_Publishers g_publishers_ForceTorqueMonitor;
+ForceTorqueMonitor_Messages g_messages_ForceTorqueMonitor;
+
+static void Ros_ForceTorqueMonitor_Initialize_Publishers(rmw_qos_profile_t const* const qos_profile);
+
+void Ros_ForceTorqueMonitor_Initialize()
+{
+    Ros_Debug_BroadcastMsg("Initializing ForceTorqueMonitor publishers");
+
+    //==================================
+    //create publishers for TCP force and external joint torque
+    const rmw_qos_profile_t* qos_profile_js = Ros_ConfigFile_To_Rmw_Qos_Profile(g_nodeConfigSettings.qos_joint_states);
+    Ros_ForceTorqueMonitor_Initialize_Publishers(qos_profile_js);
+
+}
+
+static void Ros_ForceTorqueMonitor_Initialize_Publishers(rmw_qos_profile_t const* const qos_profile)
+{
+    //create a publisher for TCP force torque
+    rcl_ret_t ret = rclc_publisher_init(
+        &g_publishers_ForceTorqueMonitor.forceTorqueTCP,
+        &g_microRosNodeInfo.node,
+        ROSIDL_GET_MSG_TYPE_SUPPORT(geometry_msgs, msg, WrenchStamped),
+        TOPIC_NAME_FORCE_TORQUE_TCP,
+        qos_profile);
+    motoRosAssert(ret == RCL_RET_OK, SUBCODE_FAIL_CREATE_PUBLISHER_JOINT_STATE_ALL);
+
+    //create message for TCP force torque
+    g_messages_ForceTorqueMonitor.forceTorqueTCP = geometry_msgs__msg__WrenchStamped__create();
+
+    char formatBuffer[MAX_JOINT_NAME_LENGTH];
+    snprintf(formatBuffer, MAX_TF_FRAME_NAME_LENGTH, "%sr%d/base", g_nodeConfigSettings.tf_frame_prefix, 1);
+    rosidl_runtime_c__String__assign(&g_messages_ForceTorqueMonitor.forceTorqueTCP->header.frame_id, formatBuffer);
+
+    //--------------
+
+    //create a publisher for joint external torque
+    ret = rclc_publisher_init(
+        &g_publishers_ForceTorqueMonitor.jointExternalTorque,
+        &g_microRosNodeInfo.node,
+        ROSIDL_GET_MSG_TYPE_SUPPORT(sensor_msgs, msg, JointState),
+        TOPIC_NAME_JOINT_EXTERNAL_TORQUE,
+        qos_profile);
+    motoRosAssert(ret == RCL_RET_OK, SUBCODE_FAIL_CREATE_PUBLISHER_JOINT_STATE_ALL);
+
+    //create message for joint external torque
+    g_messages_ForceTorqueMonitor.jointExternalTorque = sensor_msgs__msg__JointState__create();
```

Comment:
The existing `/joint_states` topic already publishes effort. Would it make sense to replace that field with these values? Or is there value in knowing the difference between internal/external torque?